3main

```diff
@@ -1332,40 +1332,35 @@ EOF
       fi
     fi
 
-    running=$(ps ax | grep -v cpuminer | grep -i screen | grep miner | awk '"miner" {print $1}')
-    if [ "$running" != "" ]
-    then
-      target=$(ps ax | grep -v cpuminer | grep -i screen | grep miner | awk '"miner" {print $1}')
-      kill "$target"
-    fi
-    sleep 2
+    while ps ax | grep SCREEN | grep -v cpuminer | grep -q miner ; do
+      echo "" | tee -a ${NVOC}/nvoc_logs/screenlog.0
+      echo "$(date) - Miner is running, 3main restarting miner" | tee -a ${NVOC}/nvoc_logs/screenlog.0
```

Comment:
@LuKePicci , You see any problems in these 2 echo?
I'm trying it on 19-2.0 , while loop kills miner, but does not echo to /home/m1/nvoc_logs/screenlog.0
Any idea?

Reply:
Got it, we have a "pkill -e miner" at the beginning of 3main, thats killing miner, not the while loop.